Untangle: Hide the header of the global sidebar on mobile

arthur791004 opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://github.com/Automattic/dotcom-forge/issues/5767

## Proposed Changes

* Add the search icon of the command palette to the masterbar on mobile behind the feature flag
* Hide the header of the global sidebar on mobile

| Before | After |
| - | - |
| ![image](https://github.com/Automattic/wp-calypso/assets/13596067/55ff33c1-92ea-47ff-814c-2b7ad1a9484d) | ![image](https://github.com/Automattic/wp-calypso/assets/13596067/819f729e-541c-4e4d-b587-fc3ea1c63ea8) |

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Go to /sites
* On the mobile view
  * Make sure you can see the search icon on the masterbar
  * Make sure the header of the global sidebar is gone

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

arthur791004 committed:
Untangle: Add the search icon of the command palette to the masterbar on mobile

arthur791004 committed:
Hide the header of the global sidebar

arthur791004 committed:
Improve layout when the viewport width is 782px

taipeicoder reviewed (approved):
LGTM! Some UI feedback:

1. The search icon has an extra right spacing. Other icons such as the hamburger and WP icon also has the same extra spacing.

![Screenshot 2024-02-26 at 5 18 18 PM](https://github.com/Automattic/wp-calypso/assets/797888/04b74bbf-1cd9-4911-ae60-c436c958f72f)

2. The hiding of the sidebar header can benefit from more polish. Users can see the header showing up. Here's the animation slowed down:

https://github.com/Automattic/wp-calypso/assets/797888/88c2b2f2-15c0-478a-9187-8b25df9f853c

3. Not related to this PR, but the Sites page behaves differently in mobile compared to the rest of the pages. In the Sites pages, the sidebar seems displayed on top, while in the other pages, the right-side content is pushed on the right. We should follow up on this cc: @lsl @davemart-in

arthur791004 committed:
Fix padding of the sidebar when sliding out

arthur791004 committed:
Fix spacing of the icon without children

arthur791004 committed:
Prevent the sidebar from displaying on the top of the sites page

arthur791004 commented:
> The search icon has an extra right spacing. Other icons such as the hamburger and WP icon also has the same extra spacing.

Addressed by https://github.com/Automattic/wp-calypso/pull/87877/commits/2aef3e7e97c73d45fac776c92796de254d994dd4

> The hiding of the sidebar header can benefit from more polish. Users can see the header showing up. Here's the animation slowed down:

Addressed by https://github.com/Automattic/wp-calypso/pull/87877/commits/fc4a08fd985ad1a105f46fcbcbbaf116eb1730ee

> Not related to this PR, but the Sites page behaves differently in mobile compared to the rest of the pages. In the Sites pages, the sidebar seems displayed on top, while in the other pages, the right-side content is pushed on the right. We should follow up on this

Addressed by https://github.com/Automattic/wp-calypso/pull/87877/commits/8a6d331a3de5d621ad169bdce3b76c0f1bb9e2b5. I think we usually push the right-side content to the right when the sidebar is visible when the viewport width is greater than 660px.

eoigal commented:
I've been looking into a bug where when we have multiple notification icons, showing multiple notification panels ends up in a broken experience. I've tried to refactor it in https://github.com/Automattic/wp-calypso/pull/87847.

I think though your PR here can help us both!.. if we don't show multiple notification icons, then we don't need that refactoring.

One suggestion (and it appears to work from my testing) is that you updated the global sidebar file - `client/layout/global-sidebar/index.jsx` with the following... I think this will help with how the header looks in the transition mentioned https://github.com/Automattic/wp-calypso/pull/87877#pullrequestreview-1900352376

```
const isDesktop = useBreakpoint( '>782px' );

return (
		<div className="global-sidebar" ref={ wrapperRef }>
			{ isDesktop && <GlobalSidebarHeader /> }
```

eoigal committed:
hide sidebar header on smaller screen sizes
